adapter/service/virtualmeter/manager.go

```diff
@@ -461,3 +444,78 @@ func (m *manager) createVirtualServicesForThing(t adapter.Thing) (outLvlSwitchSe
 
 	return nil, nil
 }
+
+// registerVirtualServices registers a virtual meter and numeric meter services for a provided thing.
+// Avoids updating if virtual service already exist.
+// VMS is always added while numeric meter is added only if virtual service is active.
+//
+//nolint:funlen,cyclop
+func (m *managerWrapper) registerVirtualServices(
+	thing adapter.Thing,
+	publisher adapter.Publisher,
+	vmsSpec *fimptype.Service,
+	numericSpec *fimptype.Service,
+) error {
+	if vmsSpec == nil || numericSpec == nil {
```

Comment:
Move this check to the invocation place in RegisterThing method. It's cleaner and more informative to see 
there if nil nil then continue instead having this hidden here. No point to log it. You would log warnings for perfectly valid things that simply don't have any level switch.